LibreHardwareMonitorLib/Hardware/Motherboard/SuperIOHardware.cs

```diff
@@ -2591,6 +2591,35 @@ private static void GetNuvotonConfigurationD(ISuperIO superIO, Manufacturer manu
 
                         break;
 
+                    case Model.ROG_MAXIMUS_Z790_HERO: //NCT6798D
+                        var knownIndices = new Dictionary<int, string>
+                        {
+                            { 0, "CPU package" },
+                            { 1, "VRM" },
+                            { 2, "Motherboard" },
+                            { 12, "Chipset" },
+                            { 21, "CPU" }
+                        };
+                        var ignoreIndices = new HashSet<int> { 3, 4, 5, 6, 7 };
+
+                        for (int i = 0; i < superIO.Temperatures.Length; i++)
+                        {
+                            if (ignoreIndices.Contains(i))
+                            {
+                                continue;
+                            }
+
+                            t.Add(new Temperature(knownIndices.ContainsKey(i) ? knownIndices[i] : $"Temperature {i}", i));
+                        }
```

Comment:
Can you follow normal structure here (no dictionary).

Reply:
Changed the code to be more explicit and inline with other code.

Reply:
Thanks!